Update docs for upgrade to React Query v3 (#424)
blitz-js/blitzjs.com: app/pages/docs/use-query.mdx

    re-render when the `data` or `error` properties change.
  - If set to `"tracked"`, access to properties will be tracked, and the
    component will only re-render when one of the tracked properties
    change.
- `notifyOnChangePropsExclusions: string[]`
  - Optional
  - If set, the component will not re-render if any of the listed
    properties change.
  - If set to `['isStale']` for example, the component will not re-render
    when the `isStale` property changes.
- `onSuccess: (data: TData) => void`
  - Optional
  - This function will fire any time the query successfully fetches new
    data.
- `onError: (error: TError) => void`
  - Optional
  - This function will fire if the query encounters an error and will be
    passed the error.
- `onSettled: (data?: TData, error?: TError) => void`
  - Optional
  - This function will fire any time the query is either successfully
    fetched or errors and be passed either the data or error
- `select: (data: TData) => unknown`
  - Optional
  - This option can be used to transform or select a part of the data
    returned by the query function.
- `suspense: boolean`
  - Optional

>   - Set this to `true` to enable suspense mode.

Reviewer: ```suggestion
  - Set this to `false` to disable suspense mode.
```

### Options {#options}

> - `queryKey: string | unknown[]`
>   - **Required**
>   - The query key to use for this query.
>   - The query key will be hashed into a stable hash. See
>     [Query Keys](../guides/query-keys) for more information.
>   - The query will automatically update when this key changes (as long as
>     `enabled` is not set to `false`).
> - `queryFn: (context: QueryFunctionContext) => Promise<TData>`
>   - **Required, but only if no default query function has been defined**
>     See [Default Query Function](../guides/default-query-function) for
>     more information.
>   - The function that the query will use to request data.
>   - Receives a `QueryFunctionContext` object with the following variables:
>     - `queryKey: QueryKey`
>   - Must return a promise that will either resolves data or throws an
>     error.

Reviewer: ```suggestion
```

We override these

- `enabled: boolean`
  - Set this to `false` to disable this query from automatically running.

>   - Can be used for [Dependent Queries](../guides/dependent-queries).

Reviewer: This and all other links on this page need updated (currently broken)

    failed query count meets that number.
- `retryOnMount: boolean`
  - If set to `false`, the query will not be retried on mount if it
    contains an error. Defaults to `true`.
- `retryDelay: number | (retryAttempt: number, error: TError) => number`
  - This function receives a `retryAttempt` integer and the actual Error
    and returns the delay to apply before the next attempt in
    milliseconds.
  - A function like
    `attempt => Math.min(attempt > 1 ? 2 ** attempt * 1000 : 1000, 30 * 1000)`
    applies exponential backoff.
  - A function like `attempt => attempt * 1000` applies linear backoff.
- `staleTime: number | Infinity`
  - The time in milliseconds after data is considered stale. This value
    only applies to the hook it is defined on.
  - If set to `Infinity`, the data will never be considered stale
- `cacheTime: number | Infinity`
  - The time in milliseconds that unused/inactive cache data remains in
    memory. When a query's cache becomes unused or inactive, that cache
    data will be garbage collected after this duration. When different
    cache times are specified, the longest one will be used.
  - If set to `Infinity`, will disable garbage collection

> - `queryKeyHashFn: (queryKey: QueryKey) => string`
>   - Optional
>   - If specified, this function is used to hash the `queryKey` to a
>     string.

Reviewer: ```suggestion
```

Don't support this
